content/data-loading.md

```diff
@@ -0,0 +1,556 @@
+  check(listId, String);
+
+  ...
+});
+```
+
+When we create a subscription to this piblication on the client, we can provide this argument via the `Meteor.subscribe()` call:
+
+```js
+Meteor.subscribe('list/todos', list._id);
+```
+
+Also note that we are careful to check the `listId` is a string as we expect, as described in the Security article.
+
+### Organizing Publications
+
+It makes sense to place a publication in a package alongside the feature that it's targeted. For instance, sometimes publications provide very specific data that's only really useful for the view that they are developed for. In that case, placing the publication in the same package as the view code makes perfect sense.
+
+Often, however, a publication is more general. For example in the Todos example application, we create a `list/todos` publication, which publishes all the todos in a list. Although in the application we only use this in one place (in the `listsShow` template), in a larger app, there's a good chance we might need to access all the todos for a list in other places. So putting the publication in the `todos` package is a sensible approach.
+
+## Using publications: Subscriptions
+
+To use publications, you need to create a subscription to it on the client. To do so, you call `Meteor.subscribe()` with the name of the publication. When you do this, it opens up a subscription to that publication, and the server starts sending data down the wire to ensure that your client collections contain up to date copies of the data that's pushed by the publication.
+
+### Organizing Subscriptions
+
+It is best to place the subscription as close as possible to the place where the data from the subscription is needed. This reduces "action at a distance" and makes it easier to understand the flow of data through your application.
+
+What this means in practice is that you should place your subscription calls in *templates*. In Blaze, it's best to do this in the `onCreated()` callback:
+
+```js
+Template.listsShowPage.onCreated(function() {
+  this.state = new ReactiveDict();
+  this.autorun(() => {
+    this.state.set('listId', FlowRouter.getParam('_id'));
+    this.subscribe('list/todos', this.state.get('listId'));
+  });
+});
+```
+
+In this code snippet we can see two important techniques for subscribing in Blaze templates:
+
+1. Calling `this.subscribe()` (rather than `Meteor.subscribe`) means that the subscription will automatically get torn down when the template is taken off the screen.
+
+2. Calling `this.autorun` sets up a reactive context which will re-initialize the subscription whenever the reactive variable `this.state.get('listId')` changes.
+
+### Fetching subscription data
+
+Once you've subscribed to a set of data, you then need to query the client collections to fetch the data again. There are a few important rules of thumb when doing this.
+
+1. Always use the same query to fetch the data from the collection that you use to publish it. 
```

Comment:
You could also use a subset of the query, but perhaps that's not really important?